inspire_json_merger/postprocess.py

```diff
@@ -0,0 +1,124 @@
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+from __future__ import absolute_import, division, print_function
+
+import itertools
+import json
+from operator import itemgetter
+
+from json_merger.conflict import Conflict
+
+from inspire_json_merger.utils import ORDER_KEY
+
+
+def postprocess_results(merged, conflicts):
+    """Run all postprocessing to provide output understandable by record-editor.
+    Args:
+        merged(dict): Merged document
+        conflicts(list): List of all possible conflicts
+
+    Returns: A tuple containing the resulted merged record in json format and a
+        an list containing all generated conflicts.
+
+    """
+    conflicts, merged = postprocess_conflicts(conflicts, merged)
+    conflicts_as_json = [json.loads(c.to_json()) for c in conflicts]
+    flat_conflicts_as_json = remove_ordering_from_authors_conflicts(
+        list(itertools.chain.from_iterable(conflicts_as_json)))
+    merged = remove_ordering_from_authors_merged(merged)
+
+    return merged, flat_conflicts_as_json
+
+
+def remove_ordering_from_authors_conflicts(conflicts):
+    """Sorts authors and cleans up ordering information from conflicts."""
+    new_conflicts = []
+    for conflict in conflicts:
+        if '/authors' in conflict["path"]:
+            del conflict["value"][ORDER_KEY]
+        new_conflicts.append(conflict)
+    return new_conflicts
+
+
+def remove_ordering_from_authors_merged(merged):
+    """Sorts authors and cleans up ordering information from merged record."""
+    if 'authors' in merged:
+        merged["authors"].sort(key=itemgetter(ORDER_KEY))
+        for author in merged.get("authors"):
+            del author[ORDER_KEY]
+    return merged
+
+
+def postprocess_conflicts(conflicts, merged):
+    """Postprocessing conflicts to display only useful conflicts.
+
+    Before flattening and serializing to JSON patch, MERGE conflict looks like this:
+    ('MANUAL_MERGE', ('authors',), ( pmap(<ROOT>), pmap(<HEAD>), pmap(<UPDATE>) )),
+    This function takes out all MANUAL_MERGE and do as follows:
+        - Ignore ROOT
+        - Update merged result with HEAD from conflict by inserting the author
+            from HEAD at the right position.
+        - Look for ADD_BACK_TO_HEAD conflict with identical content as HEAD
+            change and remove it as it's already added to HEAD
+    Args:
+        conflicts(list): List of all possible conflicts
+        merged(dict): Merged document
+
+    Returns: A tuple containing the resulted merged record in json format and a
+        an list containing all generated conflicts.
+    """
+    new_conflicts = []
+    possible_dplicates = []
+    for conflict in conflicts:
+        if conflict[0] == "MANUAL_MERGE" and "authors" in conflict[1]:
```

Comment:
isn't the check for `"authors"` too general? what about `authors` in other fields (e.g. `references`)?

Reply:
I change it to check if "authors" is first part of the first element from path. This should be unique enough .